src/Modules/KnowledgeComponents/Tutor.KnowledgeComponents.Core/Domain/KnowledgeMastery/SessionTracker.cs

```diff
@@ -27,7 +29,15 @@ public void Launch()
         if (HasUnfinishedSession)
         {
             if (IsPaused) Continue();
-            Causes(new SessionAbandoned());
+
+            var sessionAbandoned = new SessionAbandoned();
+
+            if (LastActivity.HasValue && (sessionAbandoned.TimeStamp - LastActivity.Value).TotalHours > 1)
```

Comment:
There seems to be abug here. Line 31 `if (IsPaused) Continue();` will generate a session continued event that has the current datetime, while the session abandoned might be some time in the past.

We should create the continued event here as well, making the state change comment below even more important.